### PR TITLE
front: Test E2E Rollingstock selector

### LINKED ISSUE #4104: test: Rolling stock selection
### Description and goal

- **Start page:** train creation page open

- **Steps:**
1. Click on the rolling stock selection component
2. Assert 444 results found 
3. Click on the "Electric" checkbox component
4. Assert 262 results found 
5. Click on the search bar 
6. Type "tgv"
7. Assert 27 results found
8. Click on the TGV2N2 US
9. Assert that the MR component unfolds well 
10. Click on the "Select" button
  
- **Arrival page:** train creation page open

### Acceptance criteria

- [x] The rolling stock displayed is "TGV2N2 US", standard comfort S

### PR DESCRIPTION
**Pull request by Uriel-Sautron**
close #4104 

- add test for rolling stock 
- integration_tests.yml: add artifact for playwright
- add variables.ts to be used in tests
- create front/tests/rollingstock-modal-model.ts
- use python/railjson_generator/railjson_generator/examples/rolling_stocks/fast_rolling_stock.json to create rolling stocks in tests
- use EDITOAST_URL in tests/conftest.py
- change tests/tests/test_rolling_stock.py  has been dummyfied but there is no issue with that since editoast endpoints are correctly tested

### CODE REVIEW

**Review by bloussou — COMMENTED** [review]
Can you modify the `front/README.md` to add the fact that `IS_OPEN_DATA` can be modified in `test_e2e.py` to run tests locally ?

File: tests/tests/test_rolling_stock.py [hunk 1]
```diff
@@ -1,29 +1,10 @@
...
-_FAST_ROLLING_STOCK = RollingStock(**json.loads(FAST_ROLLING_STOCK_JSON_PATH.read_text()))
+from .services import EDITOAST_URL
 
 
-@pytest.mark.usefixtures("fast_rolling_stock")
-def test_get_rolling_stocks():
-    response = requests.get(API_URL + "rolling_stock/?page_size=1000")
+def test_get_rolling_stocks(fast_rolling_stock: int):
+    response = requests.get(EDITOAST_URL + f"light_rolling_stock/{fast_rolling_stock}")
+    print(response.json())
```
Comment (bloussou) [inline tests/tests/test_rolling_stock.py]:
```suggestion
```

**Review by bloussou — CHANGES_REQUESTED** [review]
Nice !
For other reviewers : `test_rolling_stock.py` has been dummyfied but there is no issue with that since editoast endpoints are correctly tested :smile: (you can add this to the description @Uriel-Sautron for reviewers and future us)

Can you add a section in front's `README.md`to explain that to have the same setup as the CI it is required to set `CI=true` when running the test

File: tests/tests/test_rolling_stock.py (hunk 1 above)
Comment (Uriel-Sautron) [reply tests/tests/test_rolling_stock.py]: Ouppss !! Done

File: front/README.md [hunk 2]
```diff
@@ -38,12 +38,18 @@ Update licenses attributions in /src/common/ReleaseInformations/json/
...
 
 It requires:
+
 - Backend containers to be up: `docker compose build core redis api editoast postgres`
 - Running front
 - [`poetry`](https://python-poetry.org/docs/#installation) to be installed
 
-This command will execute all tests defined. The command can be changed in [`test_e2e.py`](../tests/tests/test_e2e.py) to execute only one test with or without debug mode.
-`test_e2e.py` is executing [playwright command](https://playwright.dev/docs/test-cli) in a python subprocess. Example of modification to apply to launch only `stdcm-page.spec.ts` in debug mode with chromium:
+Locally, to use the same configuration as the CI, you must change the CI variable to `CI=true`.
```
Comment (bloussou) [inline front/README.md]:
```suggestion
Locally, to use the same configuration as the CI, you must change the environment variable to `CI=true`.
```

**Review by bloussou — CHANGES_REQUESTED** [review]
Thanks, one last thing

File: front/README.md (hunk 2 above)
Comment (Uriel-Sautron) [reply front/README.md]: Done !! Thank you for your review

**Review by bloussou — APPROVED** [review]
:tada:

File: editoast/src/tests/example_rolling_stock_1.json [hunk 3]
```diff
@@ -1,91 +1,982 @@
...
-        "reference": "",
-        "family": "",
-        "type": "",
-        "grouping": "",
-        "series": "",
-        "subseries": "",
-        "unit": ""
-    },
-    "energy_sources": []
+  "version": "3.1",
```
Comment (clarani) [inline editoast/src/tests/example_rolling_stock_1.json]: Indent is not correct here, by default it is 4 spaces in the json files, can you correct this ? And also indent correctly the file :) Thanks

+ is these real data ? Are we allowed to make this data public ?

File: front/tests/assets/operationStudies/variables.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export default {
```
Comment (clarani) [inline front/tests/assets/operationStudies/variables.ts]: Can we name this file test_variables ? (suggestion)

File: front/tests/rollingstock-modal-model.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    this.getElectricalCheckbox = this.getRollingstockModal.locator('label').filter({
+      hasText: this.playwrightCommonPage.getRollingstockTranslation('electric') as string,
+    });
+  }
+
+  async openRollingstockModal() {
+    await this.getRollingStockSelector.click();
+  }
+
+  async checkResultsFound(expectedRollingstock: string) {
```
Comment (clarani) [inline front/tests/rollingstock-modal-model.ts]: (suggestion)
```suggestion
  async checkResultsNumber(expectedRollingstock: string) {
```

File: python/railjson_generator/pyproject.toml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "railjson_generator"
-version = "0.1.0"
+version = "0.1.1"
```
Comment (clarani) [inline python/railjson_generator/pyproject.toml:3]: Why do you do this change in your PR ? Do you need it ?

File: python/railjson_generator/railjson_generator/schema/infra/speed_section.py [hunk 7]
```diff
@@ -1,5 +1,5 @@
 from dataclasses import dataclass, field
-from typing import List
+from typing import List, Mapping
```
Comment (clarani) [inline python/railjson_generator/railjson_generator/schema/infra/speed_section.py:2]: Are these your change ?

File: python/railjson_generator/railjson_generator/scripts/examples/small_infra.py [hunk 8]
```diff
@@ -569,16 +569,16 @@ def add_signal_on_ports(switch, ports: Mapping[str, Tuple[str, str]]):
 # ================================
 #  Speed sections
 # ================================
-speed_0 = builder.add_speed_section(300 / 3.6)
+speed_0 = builder.add_speed_section(300 / 3.6, speed_limit_by_tag={"Divers - Haut le pied": 250 / 3.6})
```
Comment (clarani) [inline python/railjson_generator/railjson_generator/scripts/examples/small_infra.py:573]: Same here

File: tests/conftest.py [hunk 9]
```diff
@@ -111,14 +111,14 @@ def fast_rolling_stocks(request: Any) -> Iterable[int]:
...
-        requests.delete(f"{API_URL}rolling_stock/{id}/")
+        requests.delete(f"{EDITOAST_URL}rolling_stock/{id}/")
 
 
 @pytest.fixture
 def fast_rolling_stock() -> int:
     id = _create_fast_rolling_stocks()[0]
     yield id
-    requests.delete(f"{API_URL}rolling_stock/{id}/")
+    requests.delete(f"{EDITOAST_URL}rolling_stock/{id}/")
```
Comment (clarani) [inline tests/conftest.py:121]: Nice to do this ! :)

File: tests/tests/test_e2e.py [hunk 10]
```diff
@@ -45,14 +45,7 @@
...
     result = subprocess.run(
-        [
-            "yarn",
-            "--cwd",
-            "front",
-            "playwright",
-            "test",
-            "--reporter=line",
-        ],
+        ["yarn", "--cwd", "front", "playwright", "test", "--reporter=line"],
```
Comment (clarani) [inline tests/tests/test_e2e.py:48]: It's more pratical to have these in column if the linter does not complain (easier to modify to debug)!

**Review by clarani — CHANGES_REQUESTED** [review]
Left some comments ! very nice to have a new test :)

**Comment by anisometropie** [comment]
you need to squash these 

Screenshot_20230531_114420

Screenshot_20230531_114441

File: front/tests/rollingstock-modal-model.ts (hunk 5 above)
Comment (anisometropie) [inline front/tests/rollingstock-modal-model.ts]: is this the text displayed at the top of the page ?

Screenshot_20230531_115035

I would name that `checkNumberOfRollingstockFound`

File: python/railjson_generator/railjson_generator/schema/infra/speed_section.py (hunk 7 above)
Comment (Uriel-Sautron) [reply python/railjson_generator/railjson_generator/schema/infra/speed_section.py:2]: Yes, the changes here will be useful in future tests.

File: python/railjson_generator/railjson_generator/scripts/examples/small_infra.py (hunk 8 above)
Comment (Uriel-Sautron) [reply python/railjson_generator/railjson_generator/scripts/examples/small_infra.py:573]: Here too

File: python/railjson_generator/pyproject.toml (hunk 6 above)
Comment (Uriel-Sautron) [reply python/railjson_generator/pyproject.toml:3]: Change made with @bloussou

File: front/tests/assets/operationStudies/variables.ts (hunk 4 above)
Comment (Uriel-Sautron) [reply front/tests/assets/operationStudies/variables.ts]: Done !!!

File: front/tests/rollingstock-modal-model.ts (hunk 5 above)
Comment (Uriel-Sautron) [reply front/tests/rollingstock-modal-model.ts]: Done !!!

File: editoast/src/tests/example_rolling_stock_1.json (hunk 3 above)
Comment (Uriel-Sautron) [reply editoast/src/tests/example_rolling_stock_1.json]: Done !!! 
The data was retrieved from random rollingstock.

File: front/tests/rollingstock-modal-model.ts (hunk 5 above)
Comment (Uriel-Sautron) [reply front/tests/rollingstock-modal-model.ts]: Thank you good name

File: tests/tests/test_e2e.py (hunk 10 above)
Comment (Uriel-Sautron) [reply tests/tests/test_e2e.py:48]: I agree, but the linter no

**Review by Castavo — APPROVED** [review]
railjson-generator is fine, can't check the rest :grin: 

I feel like we need a way to centralize example files (namely: rolling stocks) so we don't have to maintain a bunch of them in each repo. That could make an enabler / refacto

**Review by clarani — APPROVED** [review]

**Review by leovalais — APPROVED** [review]
LGTM for the backend

File: editoast/src/tests/example_rolling_stock_1.json
Comment (flomonster) [inline editoast/src/tests/example_rolling_stock_1.json:1]: Use 4 spaces for indentation instead of tab.

**Review by flomonster — APPROVED** [review]
LGTM. Before merging please:
- Replace tabs by 4 spaces
- Rename your commits

File: editoast/src/tests/example_rolling_stock_1.json
Comment (Uriel-Sautron) [reply editoast/src/tests/example_rolling_stock_1.json:1]: Done !!

**Review by eckter — APPROVED** [review]

**Pull request merged**